changesLog.js

```diff
@@ -1,34 +1,75 @@
-  .option('-f, --filters <items>', 'comma separeted labels to filter issues', commaSeparatedList)
-  .option('-g, --groups <items>', 'comma separated labels to group by issues', commaSeparatedList)
+  .requiredOption('-m, --milestone <number>', 'must filter issue from milestone')
+  .option('-c, --customer <customer>, will filter the issues related to a customer if this option is received')
+  .option('-i, --includeAll <flag>', 'will include a section with issues done for all clients (Default: false)')
+  .option('-s, --state <state>', 'should use issues with state: closed, open, all (Default: closed)', checkValidStates)
+  .option('-f, --filters <labels>', 'comma separated labels to filter and group by issues', commaSeparatedList)
+  .option('-d, --duplicate <allow>', 'should allow duplicated issues in different groups (Default: true)')
   .parse(process.argv);
 
+let groupedIssues = [];
+const filters = commander.filters;
 const params = {
-  duplicate: commander.duplicate ? false : true,
   milestone: commander.milestone,
-  filters: commander.filters,
-  groups: commander.groups
+  state: commander.state ? commander.state : 'closed',
+  customer: commander.customer ? commander.customer : undefined,
+  duplicateIssues: commander.duplicate ? commander.duplicate : true,
+  includeIssueForAll: commander.includeAll ? commander.includeAll : true
 };
 
-console.log(`Generating Changelog for Milestone ${params.milestone}...`)
-console.log(`Duplicate issues (in different groups) set to: ${params.duplicate}.`)
-if (params.filters !== undefined) console.log(`Filters: ${params.filters}`);
-if (params.groups !== undefined) console.log(`Groups: ${params.groups}`);
+async function fetchIssues() {
+  console.log(`\n\nFetching issues in Milestone ${params.milestone}...\n`);
+  if(filters) {
+    groupedIssues = await fetchIssuesInMilestone(octokit, params, filters);
+    
+    if (groupedIssues.length > 0) {
+      console.log(`\nFinished fetching issues by filters in Milestone ${params.milestone}!\n`);
+      console.log(`Duplicate issues (in different groups) set to: ${params.duplicateIssues}.\n`);
+      console.log(`Started creating Changes log for issues based on Groups: ${filters}...\n`);
+      return true;
+    } else return false;
+  }
+  else {
+    groupedIssues = await fetchIssuesInMilestone(octokit, params);
+    
+    if (groupedIssues.length > 0) {
+      console.log(`\nFinished fecthing issues in Milestone ${params.milestone}!\n`);
+      console.log('Started creating Changes lof for issues in single block...\n');
+      return true;
+    } else return false;
+  }
+}
 
-printIssuesFromMilestone()
+async function asyncGenerateChangesLog() {
+  if(!octokit.asyncTryToken()) {
+    console.log('The token is not valid!');
+    return;
+  }
+
+  if(await fetchIssues()) {
+    console.log('\nSuccessful fetch!');
+    const changesLog = generateChangesLog(groupedIssues, params);
```

Comment:
Question: is there a reason for using `changesLog` (instead of `changeLog`)?

Reply:
Ah. For me... It seemed correct. But now I don't know. In English do we use changeLog even if there are several changes? If that's the case I might just rename all files and strings (in the end of this review). @wlthomson Just need your input in this one, please.

Reply:
English huh? 🤷‍♂ 

*changesLog* does make more logical sense - because we are talking about more than one change, so it should be plural. But.. whoever started calling it a `changeLog` went with singular on the change.

Reply:
Thank you @mark-prins I'll make this change at the end so the file renaming won't be a problem while I make all the changes for the review.